### PR TITLE
chore: Remove MVAD's dependence on hardwired credentials and azure SDKs

### PR DESCRIPTION
**Pull request by mhamilton723**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

_Briefly describe the changes included in this Pull Request._

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

AB#1944709

### CODE REVIEW

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala [hunk 1]
```diff
@@ -232,161 +237,107 @@ trait MADBase extends HasOutputCol
 
   setDefault(timestampCol -> "timestamp")
 
-  val intermediateSaveDir = new Param[String](this, "intermediateSaveDir", "Directory name " +
-    "of which you want to save the intermediate data produced while training.")
+  private def validateIntermediateSaveDir(dir: String): Boolean = {
+    assert(dir.startsWith("wasbs://"), "improper HDFS loacation. Please use a wasb path such as: \n" +
```
Comment (serena-ruan) [inline cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:241]: Is only wasbs accepted? What about abfss and say in Trident some other protocols?

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala [hunk 2]
```diff
@@ -232,161 +237,107 @@ trait MADBase extends HasOutputCol
...
-        getSASToken.get, getContainerName)
-    } else {
-      throw new Exception("You need to set either {connectionString, containerName} or " +
-        "{storageName, storageKey, endpoint, sasToken, containerName} in order to access the blob container")
-    }
-  }
+  private def getStorageInfo: StorageInfo = {
+    val uri = new URI(getIntermediateSaveDir)
+    val account = uri.getHost.split(".".toCharArray).head
+    val config = s"fs.azure.account.key.$account.blob.core.windows.net"
```
Comment (serena-ruan) [inline cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:268]: Does this config also work on databricks?

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala [hunk 3]
```diff
@@ -558,81 +501,78 @@ class DetectMultivariateAnomaly(override val uid: String) extends Model[DetectMu
...
-      val responseJson = IOUtils.toString(response.entity.get.content, "UTF-8")
-        .parseJson.asJsObject.fields
+      val summary = responseJson("summary").convertTo[DMASummary]
 
-      val summary = responseJson.get("summary").map(_.convertTo[DMASummary]).get
-      if (summary.status == "FAILED") {
+      if (summary.status.toLowerCase() == "failed") {
         val errors = summary.errors.get.toJson.compactPrint
-        throw new RuntimeException(s"Caught errors during inference: $errors")
+        throw new RuntimeException(s"Could not get trained model: $errors")
```
Comment (serena-ruan) [inline cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:552]: This is not `can't get trained model` error I guess, it should be inference failure?

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala [hunk 4]
```diff
@@ -397,6 +348,19 @@ trait MADBase extends HasOutputCol
       |""".stripMargin
   }
 
+  protected def submitDataset(dataset: Dataset[_]): Map[String, JsValue] = {
```
Comment (serena-ruan) [inline cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:351]: Can we rename this method as it does both `submitDataset` and `send request to MVAD`?

**Review by serena-ruan — COMMENTED** [review]
Thanks for all those crazy refactor!! It's super clean now and easier to maintain in the future 😄 Hooray!

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala (hunk 2 above)
Comment (mhamilton723) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:268]: yes!

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala (hunk 1 above)
Comment (mhamilton723) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:241]: Added abfss too

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala (hunk 3 above)
Comment (mhamilton723) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:552]: Fixed

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala (hunk 4 above)
Comment (mhamilton723) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala:351]: Fixed

**Pull request merged**